src/targets/gpu/jit/pointwise.cpp

```diff
@@ -91,12 +91,12 @@ struct pointwise_compiler : compiler<pointwise_compiler>
 
     compiler_replace compile(context& ctx, instruction_ref ins, const operation& op) const
     {
-        if(op.name() == "contiguous")
+        if(contains({"layout", "contiguous"}, op.name()))
         {
             return replace(compile_op(
                 ctx,
                 to_shapes(ins->inputs()),
-                {{"lambda", "[](auto x) { return x; }"}, {"kernel", "contiguous_kernel"}}));
+                {{"lambda", "[](auto x) { return x; }"}, {"kernel", op.name() + "_kernel"}}));
```

Comment:
So, `layout` op is just identity op at this point.
I am trying to understand how is this intended to work. Is the idea is to just keep it identity for now and later when MIOpen supports NHWC (if it doesn't already), we modify this `identity` to be something different ?

Reply:
Its not identity, it changes the output shape to have a different permutation.

Reply:
I see. But the buffer remains in the same order as previously. it just changes the MIGraphX shape.

Reply:
Its not the same order in memory. Of course its the same element order otherwise it would be broken. Layout doesn't do a transpose.